sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiverClient.cs

```diff
@@ -483,197 +494,696 @@ public virtual async Task<ServiceBusMessage> PeekAsync(CancellationToken cancell
+            IDictionary<string, object> propertiesToModify,
+            bool undeliverableHere)
+        {
+            Modified modified = new Modified();
+            if (undeliverableHere)
+            {
+                modified.UndeliverableHere = true;
+            }
+
+            if (propertiesToModify != null)
+            {
+                modified.MessageAnnotations = new Fields();
+                foreach (var pair in propertiesToModify)
+                {
+                    if (AmqpMessageConverter.TryGetAmqpObjectFromNetObject(pair.Value, MappingType.ApplicationProperty, out var amqpObject))
+                    {
+                        modified.MessageAnnotations.Add(pair.Key, amqpObject);
+                    }
+                    else
+                    {
+                        throw new NotSupportedException(Resources.InvalidAmqpMessageProperty.FormatForUser(pair.Key.GetType()));
+                    }
+                }
+            }
+
+            return modified;
+        }
+
+        internal Rejected GetRejectedOutcome(
+            IDictionary<string, object> propertiesToModify,
+            string deadLetterReason,
+            string deadLetterErrorDescription)
+        {
+            var rejected = AmqpConstants.RejectedOutcome;
+            if (deadLetterReason != null || deadLetterErrorDescription != null || propertiesToModify != null)
+            {
+                rejected = new Rejected { Error = new Error { Condition = AmqpClientConstants.DeadLetterName, Info = new Fields() } };
+                if (deadLetterReason != null)
+                {
+                    rejected.Error.Info.Add(ServiceBusMessage.DeadLetterReasonHeader, deadLetterReason);
+                }
+
+                if (deadLetterErrorDescription != null)
+                {
+                    rejected.Error.Info.Add(ServiceBusMessage.DeadLetterErrorDescriptionHeader, deadLetterErrorDescription);
+                }
+
+                if (propertiesToModify != null)
+                {
+                    foreach (var pair in propertiesToModify)
+                    {
+                        if (AmqpMessageConverter.TryGetAmqpObjectFromNetObject(pair.Value, MappingType.ApplicationProperty, out var amqpObject))
+                        {
+                            rejected.Error.Info.Add(pair.Key, amqpObject);
+                        }
+                        else
+                        {
+                            throw new NotSupportedException(Resources.InvalidAmqpMessageProperty.FormatForUser(pair.Key.GetType()));
+                        }
+                    }
+                }
+            }
+
+            return rejected;
+        }
+
+        /// <summary>
+        /// Returns the corresponding dead letter queue name for the queue represented by the given name.
+        /// </summary>
+        ///
+        ///<param name="queueName"></param>
+        public string GetDeadLetterQueuePath(string queueName)
```

Comment:
Ah, I see what you meant now about the helper methods.
We should probably expose this in a static class just like they did in Track 1. Maybe we can call it in `EntityNameFormatter`.